Issue4 signup functionalities

JordiGN opened the pull request:
# Description
What did you implemented/Why did you implemented this:
 - Allow user register to our application using a handed typed username, email, and password
 - Allow user to register using his/her Google Account
- The user data is registered into the Firebase application
- This is related to issue #4 
 
 Examples:
 - What?: Implemented Firebase Sign-in SDK 
 - Why?: To register the user data into the Firebase Database
 
 # Testing
Download project
NPM Install
npx pod-install (for iOS users)
npx react-native start
npx react-native run-android (Android users)
npx react-native run-ios (iOS users)

Type your name, email, and password agreed terms and conditions, and subscribe to the newsletter  
Or
Agreed terms and conditions, subscribe to the newsletter, and press sign up with Google Button
## Screenshots 
<img width="276" alt="Screen Shot 2020-11-11 at 11 56 58" src="https://user-images.githubusercontent.com/10423443/98846752-06af5880-2415-11eb-959d-82e60f9e7498.png">

<img width="296" alt="Screen Shot 2020-11-10 at 13 36 47" src="https://user-images.githubusercontent.com/10423443/98846805-19c22880-2415-11eb-9dd3-2df636202d71.png">

<img width="271" alt="Screen Shot 2020-11-11 at 11 55 15" src="https://user-images.githubusercontent.com/10423443/98846921-3bbbab00-2415-11eb-9c44-c27b2c4cb20e.png">

<img width="242" alt="Screen Shot 2020-11-11 at 12 00 34" src="https://user-images.githubusercontent.com/10423443/98847188-89381800-2415-11eb-902a-b7f490de8600.png">

tovl1 committed:
initial project

Commit:
adding iOS files

alcamk committed:
Merge pull request #5 from Brightcoders-Bootcamps/iss2-initialProject

initial project

Commit:
first approach to sign up view
co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add inputs and checks box of sign up form

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

NanciGarcia committed:
creation buttons Sign Up

Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>

NanciGarcia committed:
update style buttons

Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>

develiasdaniel committed:
add design button's of Sign Up view

Co-Authored-By: 1ng-HeberH <ing.heberh@gmail.com>
Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

Commit:
validating null form and adding or tag between buttons, also changing the touchable area to the full buttons
co-authored-by: NanciGarcia nancilizbethgarciamartines@gmail.com
co-authored-by: Oswaldotovar32 oswaldotovar32@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com

Commit:
adding the login screens

tovl1 committed:
Add disable functionality and log in tag
Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: JordiGN jordign.mx@gmail.com
Co-Authored-by: nancilizbethgarciamartinez@gmail.com

tovl1 committed:
Merge branch 'iss3-signupView' of https://github.com/Brightcoders-Bootcamps/kata-05-reservacion-vuelo-kata-05-team-02 into iss3-signupView

develiasdaniel committed:
add change color in Sign In Typing

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

NanciGarcia committed:
change color in the input password

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>

tovl1 committed:
modify styles

Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: JordiGN jordign.mx@gmail.com
Co-Authored-by: nancilizbethgarciamartinez@gmail.com

Commit:
adding simulations for loader screen

co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: NanciGaria nancilizbethgarciamartinez@gmail.com
Co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add check image in progress circular bar

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

Commit:
adding modal backgournd opacity and separating components into individual files (except terms)

C-Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
Co-Authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

NanciGarcia committed:
Creations component terms

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>

Commit:
adding an S at the end of a file name

Co-Authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
Co-Authored-by: oswaldotovar32 oswaldotovar32@gmail.com
Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com

tovl1 committed:
Merge branch 'iss3-signupView' into master

Merge iss3-signupView to master. Next merge to sandbox.

Commit:
adding package json to new brnach

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add inputs en log in form

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

NanciGarcia committed:
Cretion of mainScreen

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>

Commit:
Separating functionality for buttons, also moved the props to the mainScreen so both loginview and singupview can acces them

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdanielsanchez eliasdanielsanchezlopez@gmail.com
co-authored-by: oswaldotovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add simulated sign up or login in

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

tovl1 committed:
add function to signup in firebase

Co-Authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-Authored-by: JordiGN jordign.mx@gmail.com
Co-Authored-by: nancilizbethgarciamartinez@gmail.com

NanciGarcia committed:
restructured valid email props

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>

Commit:
adjusted the sign in function to be called just when the user email does not exist, and validating the errro messages from the sign in

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com{

NanciGarcia committed:
add function Sign Up with Count Google

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>

NanciGarcia committed:
fix conflicts

Co-Authored-By: Jordi Alejandro García Núñez <jordign.mx@hotmail.com>
Co-Authored-By: oswaldotovar32 <52464740+oswaldotovar32@users.noreply.github.com>
Co-Authored-By: Elías D Sánchez <69357302+develiasdaniel@users.noreply.github.com>

Commit:
Adding configuration files for using Google/Firebase sign in

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

develiasdaniel committed:
add configuration files for SignIn with google/firebase in android

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

Commit:
changin the bundle for GoogleService info-plist

develiasdaniel committed:
add sign in with google method

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

Commit:
adding google sign up on iOS, and also registering the data to the users table

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com

Commit:
Deleting user info when is wrong, adjusting timing for modal and laoders, validating repeated info

co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

alcamk reviewed (dismissed):
Seems good, great job team!

Commit:
Solving Code review requests

co-authored-by: NanciGarcia naniclizbethgarciamartinez@gmail.com
Co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

luisarmandodv commented on FlightReservation/components/termsComponents.js:
```diff
@@ -0,0 +1,66 @@
+      },
+    });
+    
+    export default Terms;
```
```suggestion
    export default Terms;
    
```

develiasdaniel replied:
Done!! Thanks

luisarmandodv commented on FlightReservation/components/termsComponents.js:
```diff
@@ -0,0 +1,66 @@
+  }
+
+  const styles = StyleSheet.create({
+
```
```suggestion
```

develiasdaniel replied:
Done!

luisarmandodv commented on FlightReservation/components/termsComponents.js:
```diff
@@ -0,0 +1,66 @@
+            tintColors={{ true: '#3E59F7', false: 'black' }}
+          />
+          <Text style={styles.labelCheckBox}>
+            I agree to the <Text style={{ textDecorationLine: 'underline' }}>Terms</Text> and <Text style={{ textDecorationLine: 'underline' }}>Privacy Policy.</Text>*
```
line too long, please make it shorter

develiasdaniel replied:
Ok!

luisarmandodv commented on FlightReservation/components/signupView.js:
```diff
@@ -0,0 +1,47 @@
+import Terms from './termsComponents';
+
+function SignupForm(props) {
+  const {signUpGoogle,validEmail,setValidEmail, validPassword, setValidPassword, setIsLoginFormActive,isLoginFormActive,showObj, showModal, addFill, formObjectState, modalVisible, singedText, isIconCheck} = props;
```
line too long, please make it shorter or user `Prettier`

develiasdaniel replied:
Done!

luisarmandodv commented on FlightReservation/components/signUpButtons.js:
```diff
@@ -0,0 +1,140 @@
+    },
+});
+
+export default SignUpButton;
```
```suggestion
export default SignUpButton;

```

develiasdaniel replied:
Done

luisarmandodv commented on FlightReservation/components/signUpButtons.js:
```diff
@@ -0,0 +1,140 @@
+        }
+    }
+    return (
+        <View style={[styles.containerForm, { marginTop: 20 }]}>
+            <TouchableOpacity
+                disabled={buttonDisable}
+                onPress={showModal}
+                style={[
+                    styles.ContainerOfButtonSignUp,
+                    styles.containerForm,
+                    { backgroundColor: classColor, borderWidth: 0 },
+                ]}>
+                <View>
+                    <Text style={styles.textButtons}>
+                        {isLoginFormActive ? "Log In" : "Sign Up"}
+                    </Text>
+                </View>
+            </TouchableOpacity>
+            <View style={{ alignItems: 'center', margin: 10 }}>
+                <Text style={{ color: '#C9CED6', fontSize: 15 }}>or</Text>
+            </View>
+            <TouchableOpacity disabled={buttonDisableGoogle} onPress={() => signUpGoogle()}>
+                <View
+                    style={[
+                        styles.ContainerOfButtonSignUpGoogle,
+                        { backgroundColor: classColorGoogle },
+                    ]}>
+                    <View style={styles.viewIconGoogle}>
+                        <Image
+                            style={styles.iconGoogle}
+                            source={require('../img/google.png')}
+                        />
+                    </View>
+                    <Text style={[styles.textButtons, { paddingLeft: 25 }]}>
+                        {isLoginFormActive ? "Log In" : "Sign Up"} with Google
+                    </Text>
+                </View>
```
User `Prettier` here, there are too much indentation and formating errors here.

develiasdaniel replied:
Done

luisarmandodv commented on FlightReservation/components/signUpButtons.js:
```diff
@@ -0,0 +1,140 @@
+
+function SignUpButton(props) {
+    const { objValues, showModal, setIsLoginFormActive, isLoginFormActive, signUpGoogle } = props;
+    //sign in/log in button styles
```
```suggestion
```

develiasdaniel replied:
Done!

luisarmandodv commented on FlightReservation/components/signUpButtons.js:
```diff
@@ -0,0 +1,140 @@
+    //sign in/log in button styles
+    const [classColor, setClassColor] = useState('#B6B7BA');
+    const [buttonDisable, setButtonDisable] = useState(true);
+    //sign in/log in with google button styles
```
```suggestion
```

develiasdaniel replied:
Done

luisarmandodv commented on FlightReservation/components/modalComponent.js:
```diff
@@ -0,0 +1,93 @@
+  }
+  });
+
+  export default ModalCustom;
```
```suggestion
  export default ModalCustom;
  
```

luisarmandodv commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,186 @@
+          setModalVisible(false);
+          setIsIconCheck(true);
+        }, 1000);
+        //console.log('Error: ' + e)
```
```suggestion
```

develiasdaniel replied:
Done

luisarmandodv commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,186 @@
+        setSignetText(textModal);
+        setValidEmail(true);
+        setValidPassword(true);
+        // Crea un store en Firebase
```
```suggestion
```

luisarmandodv commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,186 @@
+  }
+
+  function firebaseAuthSignUp(data) {
+    // Crea un usuario en Authentication
```
```suggestion
```

luisarmandodv commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,186 @@
+    let textModal = isLoginFormActive ? 'Logging In...' : 'Signing Up...';
+    setSignetText(textModal);
+    if (isLoginFormActive) {
+      // Accion para login
```
```suggestion
```

develiasdaniel replied:
Done

luisarmandodv reviewed (changes requested):
Remove comments, remove `console.log`, user Prettier and add `PropTypes` to all components

LeyaimMagma commented on FlightReservation/app.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "FlightReservation",
+  "displayName": "FlightReservation"
+}
```
Add blank space hera

develiasdaniel replied:
Done!

LeyaimMagma commented on FlightReservation/bdd/configFirebase.js:
```diff
@@ -0,0 +1,20 @@
+    firebase.initializeApp(firebaseConfig);
+}
+
+export { firebase };
```
Add, blank space here

develiasdaniel replied:
Done!

LeyaimMagma commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,180 @@
+    const changeSeePassword = () => {
+        setSeePassword(!isSeePassword);
+    };
+    const inputBorderColor = {
+        borderColorName: '#818181',
+        borderColorEmail: '#818181',
+        borderColorPassword: '#818181',
+    };
```
can you refactor this?

develiasdaniel replied:
can you please make a more specific suggestion?

develiasdaniel replied:
We add the colors to file const

LeyaimMagma commented on FlightReservation/components/loginView.js:
```diff
@@ -0,0 +1,40 @@
+  },
+});
+
+export default LoginView;
```
Miss Blank space

develiasdaniel replied:
Done!

fernandoamz commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,180 @@
+    const {validEmail,setValidEmail, validPassword, setValidPassword, changeForm, objValues, isLoginFormActive } = props;
+    const [isSeePassword, setSeePassword] = useState(true);
+
+    //const [validEmail, setValidEmail] = useState(true);
+    //const [validPassword, setValidPassword] = useState(true);
```
```suggestion
```

develiasdaniel replied:
Done!

fernandoamz commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,180 @@
+    const checkInputsIsNull = (e, property, propertyBorder) => {
+        const textInput = e.nativeEvent.text;
+        changeForm(property, textInput);
+        const color = !textInput ? '#818181' : '#5B6EF8';
```
All colors and theme stuff has to be inside constant file

develiasdaniel replied:
Done!

fernandoamz commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,180 @@
+            <View style={{ flexDirection: 'row' }}>
+                <Text style={styles.inputHeader}>Email</Text>
+                <Text style={validEmail ? { color: "gray" } : styles.errorText}>*</Text>
+                {!validEmail && <Text style={styles.errorText}>Email in use. Use a different email</Text>}
```
The titles and all the texts have to be inside a constant file and only import here.

develiasdaniel replied:
Done!

develiasdaniel replied:
Done!

fernandoamz commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,180 @@
+                                styles.iconSeePassword,
+                                { tintColor: borderColor.borderColorPassword },
+                            ]}
+                            source={require('../img/ojo-grey.png')}
```
All the images have to be inside the import file. you only need to import the requested file by constant

develiasdaniel replied:
Done!

fernandoamz commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,180 @@
+                </View>
+            </View>
+            <Text style={styles.instruccionsPassword}>
+                Use 8 or more characters with a mix of letters, numbers, and symbols
```
Add constants file

develiasdaniel replied:
Done!

develiasdaniel replied:
Done

fernandoamz commented on FlightReservation/components/formComponent.js:
```diff
@@ -0,0 +1,180 @@
+const styles = StyleSheet.create({
+    inputHeader: {
+        fontSize: 16,
+        color: '#818181',
```
all the colors have to be inside constants file

develiasdaniel replied:
done!

fernandoamz commented on FlightReservation/components/loginView.js:
```diff
@@ -0,0 +1,40 @@
+import ModalCustom from './modalComponent';
+
+function LoginView(props) {
+  const {signUpGoogle,validEmail,setValidEmail, validPassword, setValidPassword, setIsLoginFormActive,isLoginFormActive,showObj, showModal, addFill, formObjectState, modalVisible, singedText, isIconCheck} = props;
```
Add prettier in this line.
You can do it online. https://prettier.io/

develiasdaniel replied:
Done!

fernandoamz commented on FlightReservation/components/loginView.js:
```diff
@@ -0,0 +1,40 @@
+  return (
+    <View>
+      <Text style={styles.header}>Log In</Text>
+      <Form validEmail={validEmail} setValidEmail={setValidEmail} validPassword={validPassword} setValidPassword={setValidPassword} changeForm={addFill} objValues={formObjectState} isLoginFormActive={isLoginFormActive} />
```
Add prettier

fernandoamz commented on FlightReservation/components/loginView.js:
```diff
@@ -0,0 +1,40 @@
+  return (
+    <View>
+      <Text style={styles.header}>Log In</Text>
+      <Form validEmail={validEmail} setValidEmail={setValidEmail} validPassword={validPassword} setValidPassword={setValidPassword} changeForm={addFill} objValues={formObjectState} isLoginFormActive={isLoginFormActive} />
```
```suggestion
      <Form
          validEmail={validEmail}
          setValidEmail={setValidEmail}
          validPassword={validPassword}
          setValidPassword={setValidPassword}
          changeForm={addFill}
          objValues={formObjectState}
          isLoginFormActive={isLoginFormActive} 
       />
```

develiasdaniel replied:
Done!

fernandoamz commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,186 @@
+
+import SignupForm from './signupView';
+import LoginView from './loginView';
+// import {
+//   GoogleSignin,
+//   GoogleSigninButton,
+//   statusCodes,
+// } from 'react-native-google-signin';
```
```suggestion

```

fernandoamz commented on FlightReservation/components/mainScreen.js:
```diff
@@ -0,0 +1,186 @@
+import { GoogleSignin } from '@react-native-community/google-signin';
+
+GoogleSignin.configure({
+  webClientId: '461524927651-kr6p5l2toffdi6sqgs8tktcj8v01nvjj.apps.googleusercontent.com',
```
Add to `.env` file this string

develiasdaniel replied:
DONE! thank you

develiasdaniel replied:
Done!! thank you

fernandoamz reviewed (changes requested):
by last I recommend you read this blog. It would be nice to improve your legibility on the code.

https://dev.to/carlillo/clean-code-applied-to-javascript-part-i-before-your-start-16ic

Commit:
solving changes requested on Pull request #5

Co-authored-by: NanciGarcia nancilizbethgarciamartinez@gmail.com
Co-authored-by: develiasdaniel eliasdanielsanchezlopez@gmail.com
Co-authored-by: OswaldoTovar32 oswaldotovar32@gmail.com

Commit:
merging issue5 and issue4 into a temporal branch in order to solve merge conlicts

Co-authorerd-by: NanciGarcia nancilizbethgarciamartienz@gmail.com
Co-authorerd-by:OswaldoTovar32 oswaldotovar32@gmail.com
Co-authorerd-by: develiasdaniel eliasdanielsanchezlopez@gmail.com

Ocosmee reviewed (dismissed):
Everything Ok, only if you don't use a function on signUpView  'seepass()' you should deleted

develiasdaniel committed:
refactoring changes suggested in the pr #4

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

develiasdaniel committed:
add changes suggested in pr #4

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>

develiasdaniel committed:
add prop types

Co-Authored-By: oswaldotovar32 <oswaldotovar32@gmail.com>
Co-Authored-By: NanciGarcia <nancilizbethgarciamartinez@gmail.com>
Co-Authored-By: JordiGN <jordign.mx@gmail.com>
